### ISSUE TITLE
🐛 `useSortedKeys` should allow case-sensitivity to be configured

### ISSUE THREAD

**Issue by ek68794998** [biomejs/biome#8981]
### Environment information

<details>

```bash
LI:
  Version:                      2.3.11
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.15.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.10.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0
```

</details>

### What happened?

1. Configure a project with ESLint:
    ```
	"sort-keys": [
		"error",
		"asc",
		{
			allowLineSeparatedGroups: true,
			caseSensitive: false,
			natural: true,
		},
	],
    ```
2. Ensure that `useSortedKeys` is enabled in the `assist` section of `biome.json`.
3. Add the following code to any function:
    ```
	return {
		iconBgClassName,
		IconComponent,
		iconFgClassName,
	};
    ```

Biome will complain about the order (IDE warnings from Zed); upon saving the file, it will move `IconComponent` up one line, which will upset ESLint; if ESLint auto-fix is enabled, it will move `IconComponent` back down, which brings back the Biome IDE warning.

---

**More detail:**

The reason I've filed this as a bug and not as a feature request is because the way this behavior is documented is incorrect, and this also appears to be a feature parity issue with [`sort-objects`, used as a rule source](https://github.com/biomejs/biome/blob/cc7a4785f47a2c2a5ba3ea4ef90937742a2cd347/crates/biome_js_analyze/src/assist/source/use_sorted_keys.rs#L137).

According to [the Biome documentation](https://next.biomejs.dev/assist/actions/use-sorted-keys/#description):

> Sort the keys of a JSON object in natural order.
> 
> [Natural order](https://en.wikipedia.org/wiki/Natural_sort_order) means that uppercase letters come before lowercase letters (e.g. A < a < B < b) and numbers are compared in a human way (e.g. 9 < 10).

However, natural sort order **does not necessarily mean** that things should be sorted case-sensitively; in fact, the linked Wikipedia article states (emphasis mine):

> Natural sorting algorithms generally operate by splitting strings into "chunks" of text and numbers.
> 
> - Text chunks are compared alphabetically (**often case-insensitively**).
> - Numeric chunks are parsed into integer values and compared numerically.

The [aforementioned Perfectionist rule on which this was based](https://perfectionist.dev/rules/sort-objects) supports `ignoreCase` as a configuration value.

The biggest problem with this is that it competes with ESLint configurations (using either `sort-objects` or [`sort-keys`](https://eslint.org/docs/latest/rules/sort-keys)), which means an inline comment is required to reconcile the two sorts:

<img width="1987" height="594" alt="Image" src="https://github.com/user-attachments/assets/e2f6094e-dce8-491b-8beb-5f3c46e73b12" />

### Expected result

`useSortedKeys` should be configurable so that it ignores case and can be reconciled with ESLint's `sort-keys` rule.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
A note about your filing, the assist action **is inspired**, which means that we diverged intentionally from the original rule.

That doesn't mean your issue isn't valid. Thank you for the submission!
